src/IIIFPresentation/API/Features/Storage/Requests/GetHierarchicalCollection.cs

```diff
@@ -136,21 +143,36 @@ INNER JOIN
         }
 
         List<Collection>? items = null;
+        string? collectionFromS3 = null;
 
         if (collection != null)
         {
-            items = await dbContext.Collections
-                .Where(s => s.CustomerId == request.CustomerId && s.Parent == collection.Id)
-                .ToListAsync(cancellationToken: cancellationToken);
+            if (!collection.IsStorageCollection)
+            {
+                var objectFromS3 = await bucketReader.GetObjectFromBucket(new ObjectInBucket(settings.S3.StorageBucket,
+                    $"{request.CustomerId}/collections/{collection.Id}"), cancellationToken);
 
-            foreach (var item in items)
+                if (objectFromS3.Stream != null)
+                {
+                    StreamReader reader = new StreamReader(objectFromS3.Stream);
+                    collectionFromS3 = reader.ReadToEnd();
+                }
+            }
+            else
             {
-                item.FullPath = collection.GenerateFullPath(item.Slug);
+                items = await dbContext.Collections
+                    .Where(s => s.CustomerId == request.CustomerId && s.Parent == collection.Id)
+                    .ToListAsync(cancellationToken: cancellationToken);
+
+                foreach (var item in items)
```

Comment:
Since `items` is a list, you could also do `items.ForEach(item => item.FullPath = collection.GenerateFullPath(item.Slug));`